gatsby-node.js

```diff
@@ -4,35 +4,40 @@ const deepEqual = require('deep-equal');
 
 /**
  * Fetches all records for the current index from Algolia
- *
- * @param {AlgoliaIndex} index eg. client.initIndex('your_index_name');
- * @param {Array<String>} attributesToRetrieve eg. ['modified', 'slug']
  */
-function fetchAlgoliaObjects(
-  index,
-  attributesToRetrieve = ['modified'],
-  reporter
-) {
+function fetchExistingObjects(index, reporter, cache) {
   const hits = {};
 
-  return index
-    .browseObjects({
-      batch: batch => {
-        if (Array.isArray(batch)) {
-          batch.forEach(hit => {
-            hits[hit.objectID] = hit;
-          });
-        }
-      },
-      attributesToRetrieve,
+  return cache
+    .get('algolia-objects')
+    .then(values => {
+      if (!values) {
+        throw new Error('cache actually failed');
+      }
```

Comment:
Wouldn't empty cache valides this condition? Or is `values` an array?

Reply:
The cache gets written based on the result of the graphql query, so indeed if that would be empty `{}` would be in the cache, and also should be invalidated